src/runtime_src/core/pcie/driver/linux/xocl/subdev/flash.c

```diff
@@ -467,6 +471,9 @@ static int flash_transaction(struct xocl_flash *flash,
 		ret = flash_rx(flash, NULL, len);
```

Comment:
we might want to ignore the return code when we just want to drain the fifo here.

Reply:
Yes Max, I have tested the flash with export FLASH_VIA_DRIVER=1. And it is working fine.
So you are asking we should only check the return for  "need_output" case. 
I will make the change.